packages/graphql/src/schema/validation/custom-rules/features/valid-global-id.ts

```diff
@@ -0,0 +1,277 @@
+                    );
+                    if (!isValid) {
+                        const errorOpts = {
+                            nodes: [objectType, fieldNamedID],
+                            path: [objectType.name.value, ...errorPath],
+                            source: undefined,
+                            positions: undefined,
+                            originalError: undefined,
+                        };
+
+                        // TODO: replace constructor to use errorOpts when dropping support for GraphQL15
+                        context.reportError(
+                            new GraphQLError(
+                                errorMsg || "Error",
+                                errorOpts.nodes,
+                                errorOpts.source,
+                                errorOpts.positions,
+                                errorOpts.path,
+                                errorOpts.originalError
+                            )
+                        );
+                    }
+                },
+            },
+
+            InterfaceTypeDefinition: {
+                leave(interfaceType: InterfaceTypeDefinitionNode) {
+                    addToAliasedFieldsMap(interfaceType.name.value);
+                    const fieldNamedID = getUnaliasedFieldNamedID(interfaceType);
+                    if (
+                        !fieldNamedID ||
+                        !hasGlobalIDField(interfaceType, typeNameToGlobalId, interfaceToImplementingTypes)
+                    ) {
+                        return;
+                    }
+
+                    const inheritedAliasedFields = (
+                        getInheritedTypeNames(interfaceType, interfaceToImplementingTypes) || []
+                    ).map(getAliasedFieldsFromMap);
+
+                    const { isValid, errorMsg, errorPath } = assertValid(
+                        assertGlobalIDDoesNotClash.bind(null, inheritedAliasedFields)
+                    );
+                    if (!isValid) {
+                        const errorOpts = {
+                            nodes: [interfaceType],
+                            path: [interfaceType.name.value, ...errorPath],
+                            source: undefined,
+                            positions: undefined,
+                            originalError: undefined,
+                        };
+
+                        // TODO: replace constructor to use errorOpts when dropping support for GraphQL15
+                        context.reportError(
+                            new GraphQLError(
+                                errorMsg || "Error",
+                                errorOpts.nodes,
+                                errorOpts.source,
+                                errorOpts.positions,
+                                errorOpts.path,
+                                errorOpts.originalError
+                            )
+                        );
+                    }
+                },
```

Comment:
This looks as far as I can tell, to be identical to above - shared function?

Reply:
The function looks long because of the GraphQL error but in fact there are only 3 instructions that are repeated. 
I wouldn't introduce another function just for that, it would require 4 arguments and it does not provide anything useful. 
If anything, if we change the interface behaviour we will need to discard this function to modify it..